src/validate_incoming.rs

```diff
@@ -0,0 +1,253 @@
+use crate::MAuthInfo;
+use chrono::prelude::*;
+use mauth_core::verifier::Verifier;
+use reqwest::{Client, Method, Request};
+use thiserror::Error;
+use uuid::Uuid;
+
+/// This struct holds the app UUID for a validated request. It is meant to be used with the
+/// Extension setup in Hyper requests, where it is placed in requests that passed authentication.
+/// The custom struct makes it clearer that the request has passed and this is an authenticated
+/// app UUID and not some random UUID that some other component put in place for some other
+/// purpose.
+#[derive(Debug, Clone)]
+#[non_exhaustive]
+pub struct ValidatedRequestDetails {
+    pub app_uuid: Uuid,
+}
+
+impl MAuthInfo {
+    pub(crate) async fn validate_request(
```

Comment:
It would be good to have some test.

These validate functions require an Axum Request, while the signing requests require a reqWest Request.
I know reqWest is the most popular high level client library, not sure about Axum as the community (last time I checked 1 year ago) is fragmented.

Potentially we could use http:request everywhere but I do not know how much of PITA that would be.

Reply:
I found it impossible to build a version based on the http crate's request trait because there is no interface provided to allow you to read the contents of the body without consuming it and making it unavailable to the rest of the stack. Constructing a new body with the request data is also not possible, as the concrete type would be unknown and there is no method on the trait to create a new instance. If you can figure out a way to do this, I'd love to know!

Axum is not a dominantly popular Rust web framework. However, to the best of my knowledge, it is the only one integrated with the Tower ecosystem that makes it at least theoretically possible to build reusable and composable middleware like this. To have a working request verifier with any other currently existing Rust web framework would require a complete rewrite anyways to use the interfaces provided by that specific web framework.